docs/src/pages/documentation/cloudflare-workers.md

```diff
@@ -0,0 +1,375 @@
+import { FragmentGateway, getWebMiddleware } from "web-fragments/gateway";
+import type { Fetcher } from "@cloudflare/workers-types";
+
+export interface Env {
+	ASSETS: {
+		fetch: (request: Request) => Promise<Response>;
+	};
+	FRAGMENT_SERVICE: Fetcher;
+}
+
+export default {
+	async fetch(request: Request, env: Env, _ctx) {
+		// Create the gateway inside the fetch handler
+		// This ensures we have access to env bindings
+		const gateway = new FragmentGateway();
+
+		// Register your fragment
+		gateway.registerFragment({
+			fragmentId: "my-fragment",
+			// The endpoint is your Service Binding's fetch method
+			endpoint: env.FRAGMENT_SERVICE.fetch.bind(env.FRAGMENT_SERVICE),
+			routePatterns: [
+				// When users visit /dashboard in the shell app, route to this fragment
+				// Note: The fragment's internal routes don't need to match this path
+				// For example, your fragment might only have a "/" route, but the gateway
+				// mounts it at /dashboard. The fragment uses basename config to handle this.
+				"/dashboard/:_*",
+				// Handle fragment assets (the :_* means "match everything after")
+				"/__wf/my-fragment.workers.dev/:_*",
+				// Handle any assets your fragment requests
+				"/assets/:_*",
+			],
+			// Only forward headers that your fragment needs
+			// This reduces payload size and improves security
+			forwardFragmentHeaders: ["accept-language", "user-agent", "cookie"],
+		});
+
+		// Get the Web Fragments middleware
+		const middleware = getWebMiddleware(gateway, {
+			mode: "production",
+		});
+
+		// Handle CORS preflight if needed
+		if (request.method === "OPTIONS") {
+			return new Response(null, {
+				headers: {
+					"Access-Control-Allow-Origin": "*",
+					"Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
+					"Access-Control-Allow-Headers": "Content-Type, x-fragment-mode, x-web-fragment-id",
+				},
+			});
+		}
+
+		// Let the middleware handle routing
+		return middleware(request, async () => {
+			// Fallback: serve your shell application assets
+			// This runs when the request doesn't match any fragment routes
+			return env.ASSETS.fetch(request);
+		});
+	},
+};
+```
+
+#### Understanding Route Patterns
+
+The `routePatterns` array tells the gateway which URLs should be routed to your fragment:
+
+- **Application routes**: `/dashboard/:_*` matches `/dashboard` and any nested paths
+- **Asset routes**: `/__wf/my-fragment.workers.dev/:_*` is the unique prefix for your fragment's assets
+- **Wildcard**: The `:_*` syntax means "match everything that follows"
+
+Think of route patterns as traffic signs for your gateway...they tell it where to send each request.
+
+**Important**: Your fragment worker must be configured to handle the path where the gateway mounts it. If the gateway routes `/dashboard/:_*` to your fragment, your fragment needs to expect requests at `/dashboard`. See your framework's documentation for handling base paths (e.g., Remix's `basename` option).
```

Comment:
I don't think that using `basename` is the best way to do this, because you might want to configure a redirect from `/` to `/dashboard/` in your root route which would not be possible if you configured the basepath.

additionally, if you want to document this, maybe there should be a section higher up on what we need the remix config to be, which essentially comes down to setting assetDir to be `__wf/my-fragment/`, configuring the `/dashboard` route, and optinoally configuring the redirect to `/dashboard` I mentioned above.